### PR TITLE
bug(clerk-js): Use EIP6963 standard to get web3 provider

### PR DESCRIPTION
**Pull request by EmmanouelaPothitou**
## Description
Use the EIP-6963 standard to get a Web3 provider when more than one provider is injected.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/utils/web3.ts [hunk 1]
```diff
@@ -27,15 +30,16 @@ type GenerateWeb3SignatureParams = {
...
-export async function generateWeb3Signature({ identifier, nonce }: GenerateWeb3SignatureParams): Promise<string> {
-  // @ts-ignore
-  if (!global.ethereum) {
-    // Do nothing when ethereum doesn't exist.
+export async function generateWeb3Signature(params: GenerateWeb3SignatureParams): Promise<string> {
+  const { identifier, nonce, provider } = params;
+  const injectedProvider = injectedWeb3Providers.get(provider);
+  if (!injectedProvider) {
+    // TODO: discuss what happens if a plugin for the requested provider is not found in the system
+    //do we simply return '' or fall back to another plugin?
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/utils/web3.ts:38]: Ditto

**Review by nikosdouvlis — APPROVED** [review]

File: packages/clerk-js/src/utils/web3.ts [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+// global.etherem -> coinbase
+// injectedProvider(metamask), will use metamask wallet
+// generateWeb3Signature(), will use coinbase wallet
+//                      neds to be generateWeb3Signature({provider: metamask}) instead
+
+export async function getWeb3Identifier(params: GetWeb3IdentifierParams) {
+  const injectedProvider = injectedWeb3Providers.get(params.provider);
+  if (!injectedProvider) {
+    // TODO: discuss what happens if a plugin for the requested provider is not found in the system
+    //do we simply return '' or fall back to another plugin?
```
[the remark this answers is not in the document]
Comment (chanioxaris) [reply packages/clerk-js/src/utils/web3.ts:19]: As for now it's fine to not do anything and let the flow fail, as it was working before

[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply packages/clerk-js/src/utils/web3.ts]: Alright, then @EmmanouelaPothitou please update the comments to indicate what Haris said

File: .changeset/four-tips-push.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@clerk/clerk-js": patch
+---
+
+Use the EIP-6963 standard to get a Web3 provider when more than one provider is injected.
```
Comment (panteliselef) [inline .changeset/four-tips-push.md:5]: 🙃
```suggestion
Bug fix: Use the EIP-6963 standard to get a Web3 provider when more than one provider is injected.
```

**Review by panteliselef — APPROVED** [review]

**Review by chanioxaris — APPROVED** [review]
💯

**Pull request merged**